File: ad_library/src/main/java/com/adgrowth/adserver/exceptions/AdRequestException.java
package com.adgrowth.adserver.exceptions;

public class AdRequestException extends Exception {
    public final static String PLAYBACK_ERROR = "playback_error";
    public final static String NETWORK_ERROR = "network_error";
    public final static String UNKNOWN_ERROR = "unknown_error";
    public final static String UNIT_ID_MISMATCHED_AD_TYPE = "unit_id_mismatched_ad_type";
    public final static String INTERNAL_ERROR = "internal_error";
    public final static String INVALID_CLIENT_KEY = "invalid_client_key";
    public final static String INVALID_UNIT_ID = "invalid_unit_id";
    public final static String SDK_NOT_INITIALIZED = "sdk_not_initialized";
    public final static String NO_AD_FOUND = "no_ad_found";
    public static final String ALREADY_LOADED = "already_loaded";
    private String mCode;

    public AdRequestException(String code) {
        this.mCode = code;
    }

    public String getCode() {
        return mCode;
    }

}

File: ad_library/src/main/kotlin/com/adgrowth/adserver/exceptions/AdRequestException.kt
package com.adgrowth.adserver.exceptions

class AdRequestException(val code: String) : Exception() {

    companion object {
        const val PLAYBACK_ERROR = "playback_error"
        const val NETWORK_ERROR = "network_error"
        const val UNKNOWN_ERROR = "unknown_error"
        const val UNIT_ID_MISMATCHED_AD_TYPE = "unit_id_mismatched_ad_type"
        const val INTERNAL_ERROR = "internal_error"
        const val INVALID_CLIENT_KEY = "invalid_client_key"
        const val INVALID_UNIT_ID = "invalid_unit_id"
        const val SDK_NOT_INITIALIZED = "sdk_not_initialized"
        const val NO_AD_FOUND = "no_ad_found"
        const val APP_NOT_FOUND = "app_not_found"
        const val ALREADY_LOADED = "already_loaded"
        const val ALREADY_CONSUMED = "already_consumed"
        const val NOT_READY = "not_ready"
        const val ALREADY_SHOWING_FULL_SCREEN_AD = "already_showing_full_screen_ad"
    }
}
